### PR TITLE
Fixcla

### PR DESCRIPTION
**Pull request by xwz123**
1.  add check cla whe occur source branch changed.
2.  check event payload and modify get owner and repo by common method
3. fix use pr.Head.user as the pr author

### CODE REVIEW

File: prow/gitee-plugins/cla-euler/cla.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"k8s.io/test-infra/prow/gitee"
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla-euler/cla.go]: move this code below line 15

File: prow/gitee-plugins/cla-euler/cla.go [hunk 2]
```diff
@@ -67,37 +68,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
 		log.WithField("duration", time.Since(funcStart).String()).Debug("Completed handleNoteEvent")
 	}()
 
-	if *(e.Action) != "comment" {
+	ne := gitee.NewNoteEventWrapper(e)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla-euler/cla.go:74]: ne := gitee.NewPRNoteEvent(e)

File: prow/gitee-plugins/cla-euler/cla.go [hunk 3]
```diff
@@ -67,37 +68,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
...
 	// Only consider "/check-cla" comments.
-	if !checkCLARe.MatchString(e.Comment.Body) {
+	if !checkCLARe.MatchString(ne.GetComment()) {
 		return nil
 	}
 
-	pr := e.PullRequest
-	org := e.Repository.Namespace
-	repo := e.Repository.Path
+	return cl.handlePullRequestComment(gitee.NewPRNoteEvent(e), log)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla-euler/cla.go:89]: return cl.handlePullRequestComment(ne, log)

File: prow/gitee-plugins/cla-euler/cla.go [hunk 4]
```diff
@@ -67,37 +68,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
...
+}
 
+func (cl *cla) handlePullRequestComment(e gitee.PRNoteEvent, log *logrus.Entry) error {
+	org, repo := gitee.GetOwnerAndRepoByEvent(e.NoteEvent)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla-euler/cla.go:93]: org, repo := e.GetOrgRepo()

File: prow/gitee-plugins/cla-euler/cla.go [hunk 5]
```diff
@@ -142,19 +145,13 @@ func (cl *cla) handlePullRequestEvent(e *sdk.PullRequestEvent, log *logrus.Entry
 		log.WithField("duration", time.Since(funcStart).String()).Debug("Completed handlePullRequest")
 	}()
 
-	if e.PullRequest.State != "open" {
-		log.Debug("Pull request state is not open, skipping...")
-		return nil
-	}
-
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla-euler/cla.go:149]: why should this be deleted?

File: prow/gitee-plugins/cla/cla.go [hunk 6]
```diff
@@ -66,38 +67,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
 	defer func() {
 		log.WithField("duration", time.Since(funcStart).String()).Debug("Completed handleNoteEvent")
 	}()
-
-	if *(e.Action) != "comment" {
+	ne := gitee.NewNoteEventWrapper(e)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla/cla.go:72]: ditto

File: prow/gitee-plugins/cla/cla.go [hunk 7]
```diff
@@ -66,38 +67,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
...
 	// Only consider "/check-cla" comments.
-	if !checkCLARe.MatchString(e.Comment.Body) {
+	if !checkCLARe.MatchString(ne.GetComment()) {
 		return nil
 	}
 
-	pr := e.PullRequest
-	org := e.Repository.Namespace
-	repo := e.Repository.Path
+	return cl.handlePullRequestComment(gitee.NewPRNoteEvent(e), log)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla/cla.go:87]: ditto

File: prow/gitee-plugins/cla/cla.go [hunk 8]
```diff
@@ -66,38 +67,40 @@ func (cl *cla) handleNoteEvent(e *sdk.NoteEvent, log *logrus.Entry) error {
...
+}
 
+func (cl *cla) handlePullRequestComment(e gitee.PRNoteEvent, log *logrus.Entry) error {
+	org, repo := gitee.GetOwnerAndRepoByEvent(e.NoteEvent)
```
Comment (zengchen1024) [inline prow/gitee-plugins/cla/cla.go:91]: ditto

File: prow/gitee-plugins/dispatcher.go [hunk 9]
```diff
@@ -10,9 +10,10 @@ import (
 	"sync"
 	"time"
 
+	hook "k8s.io/test-infra/prow/gitee-hook"
```
Comment (zengchen1024) [inline prow/gitee-plugins/dispatcher.go]: why should change this?

File: prow/gitee-plugins/util.go [hunk 10]
```diff
@@ -166,3 +169,58 @@ func convertPullRequestLabel(e *gitee.PullRequestEvent) []github.Label {
...
 }
+
+func checkNoteEvent(e *gitee.NoteEvent) error {
+	eventType := "note event"
+	ne := pgitee.NewNoteEventWrapper(e)
+	if ne.Comment == nil {
+		return fmtCheckError(eventType, "comment")
+	}
+	if ne.IsPullRequest() {
+		return checkPullRequestHook(ne.PullRequest, eventType)
```
Comment (zengchen1024) [inline prow/gitee-plugins/util.go:180]: if return directly, line 185 will not run

File: prow/gitee-plugins/util.go [hunk 11]
```diff
@@ -166,3 +169,58 @@ func convertPullRequestLabel(e *gitee.PullRequestEvent) []github.Label {
...
+	}
+	if ne.IsIssue() && ne.Issue == nil {
+		return fmtCheckError(eventType, "issue")
```
Comment (zengchen1024) [inline prow/gitee-plugins/util.go:183]: ditto

File: prow/gitee/util.go [hunk 12]
```diff
@@ -4,6 +4,7 @@ import (
 	sdk "gitee.com/openeuler/go-gitee/gitee"
 )
 
+//GetOwnerAndRepoByEvent obtain the owner and repository name from the event
```
Comment (zengchen1024) [inline prow/gitee/util.go:7]: it seems this function is useless, it is better to define two fucntions which fetch org and repo from PullRequestEvent and IssueEvent respectively

File: prow/gitee-plugins/util.go [hunk 13]
```diff
@@ -1,8 +1,11 @@
 package plugins
 
 import (
+	"fmt"
 	"strings"
 
+	pgitee "k8s.io/test-infra/prow/gitee"
```
Comment (zengchen1024) [inline prow/gitee-plugins/util.go:7]: "k8s.io/test-infra/prow/gitee"

sdk "gitee.com/openeuler/go-gitee/gitee"

**Review by zengchen1024 — COMMENTED** [review]
need change

File: prow/gitee-plugins/cla-euler/cla.go (hunk 1 above)
Comment (xwz123) [reply prow/gitee-plugins/cla-euler/cla.go]: aready fix this on [0e1c184](https://github.com/opensourceways/test-infra/pull/57/commits/0e1c18461842377f58b973218d59656b5f1025c9)

File: prow/gitee-plugins/cla-euler/cla.go (hunk 5 above)
Comment (xwz123) [reply prow/gitee-plugins/cla-euler/cla.go:149]: Misoperation

File: prow/gitee-plugins/dispatcher.go (hunk 9 above)
Comment (xwz123) [reply prow/gitee-plugins/dispatcher.go]: [go imports](https://github.com/golang/go/wiki/CodeReviewComments#imports)

File: prow/gitee-plugins/util.go (hunk 11 above)
Comment (xwz123) [reply prow/gitee-plugins/util.go:183]: if a not event triggered by issue comment and issue field is empty this event should not be handled

File: prow/gitee-plugins/util.go (hunk 13 above)
Comment (xwz123) [reply prow/gitee-plugins/util.go:7]: Combining [go imaports](https://github.com/golang/go/wiki/CodeReviewComments#imports), I think the local package should be aliased first, but I will modify it for consistency

File: prow/gitee/util.go (hunk 12 above)
Comment (xwz123) [reply prow/gitee/util.go:7]: i'll submit a PR to fix this issue

File: prow/gitee/util.go [hunk 14]
```diff
@@ -4,29 +4,12 @@ import (
...
 }
-//GetOwnerAndRepoByPRBranch obtain the owner and repository name from the pullrequest's branch
-func GetOwnerAndRepoByPRBranch(prb *sdk.BranchHook) (string, string) {
-	if prb.Repo == nil {
-		return "", ""
-	}
-	return prb.Repo.Namespace, prb.Repo.Path
+
+//GetOwnerAndRepoByIssueEvent obtain the owner and repository name from the issue's event
+func GetOwnerAndRepoByIssueEvent(pre *sdk.PullRequestEvent) (string, string) {
```
Comment (zengchen1024) [inline prow/gitee/util.go:13]: the parameter is wrong

**Review by zengchen1024 — COMMENTED** [review]
need change

**Pull request merged**